lib/snapshot.go

```diff
@@ -664,10 +664,6 @@ func (snap *Snapshot) RemoveChecksumBytes(key []byte, value []byte) {
 // WaitForAllOperationsToFinish will busy-wait for the snapshot channel to process all
 // current operations. Spinlocks are undesired but it's the easiest solution in this case,
 func (snap *Snapshot) WaitForAllOperationsToFinish() {
-	// If we haven't hit the first snapshot block height yet, don't bother.
-	if !snap.shouldPerformSnapshotOperations() {
```

Comment:
This was a tricky one. We only call this function in two places right now, but the right thing to do is actually to always wait for the operation queue to empty, otherwise you will may have checksum operations that you haven't processed yet. This actually messed up the compute_db_checksum.go script, which was super annoying because it relied on this call. Super annoying to find but eventually I realized the values were different every time depending on what operations it had processed lol...